R/binary_classification.R

```diff
@@ -138,3 +138,210 @@ fbeta_score <- function(actual, predicted, beta = 1) {
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' fnr(actual, predicted)
+
+fnr <- function(actual, predicted) {
+    1 - sensitivity(actual, predicted)
+}
+
+#' False Positive Rate
+#'
+#' \code{fnr} calculates the proportion of actual negatives
+#' that are not identified as such.
+#' It is defined as \code{1 - specificity}
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{specificity}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' fpr(actual, predicted)
+
+fpr <- function(actual, predicted) {
+    1 - specificity(actual, predicted)
+}
+
+#' Positive Predictive Value
+#'
+#' \code{ppv} calculates the proportion positive values that
+#' are true positive results.
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{npv}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' ppv(actual, predicted)
+
+ppv <- function(actual, predicted) {
+    cm <- cmat(actual, predicted)
+    cm$tp / (cm$tp + cm$fp)
+}
+
+#' Negative Predictive Value
+#'
+#' \code{ppv} calculates the proportion negative values that
+#' are true negative results.
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{npv}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' npv(actual, predicted)
+npv <- function(actual, predicted) {
+    cm <- cmat(actual, predicted)
+    cm$tn / (cm$tn + cm$fn)
+}
+
+#' False Discovery Rate
+#'
+#' \code{fdr} is the complement of the Positive Predictive
+#' Value (\code{ppv}), and is the proportion of positive
+#' results that are false positives.
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{ppv}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' fpr(actual, predicted)
+
+fdr <- function(actual, predicted) {
+    1 - ppv(actual, predicted)
+}
+
+#' False Omission Rate
+#'
+#' \code{fomr} is the complement of the Negative Predictive
+#' Value (\code{npv}), and is the proportion of negative
+#' results that are false negatives.
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{cmat}} \code{\link{npv}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' fomr(actual, predicted)
+
+fomr <- function(actual, predicted) {
+    1 - npv(actual, predicted)
+}
+
+#' Positive Likelihood Ratio
+#'
+#' \code{lrp} is used to assessing the value of performing a
+#' diagnostic test, and estimates the ratio of the probability
+#' of a true positive result over the probability of a false positive
+#' result: \code{sensitivity / (1 - specificity)}
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{tpr}} \code{\link{fpr}}
+#' @export
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' lrp(actual, predicted)
+
```

Comment:
In all of these functions, can you remove the extra white line between the documentation and the function definition? Or, is this a roxygen best practice that I'm not aware of?

Reply:
Done, removed extra line